apps/minifront/src/components/send/send-form/index.tsx

```diff
@@ -31,11 +35,42 @@ export const SendForm = () => {
     setFeeTier,
     setMemo,
     sendTx,
+    setGasPrices,
+    setStakingToken,
     txInProgress,
   } = useStore(sendSelector);
 
   useRefreshFee();
 
+  // Determine if the selected token is the staking token based on the current balances and metadata
+  const isStakingToken = hasStakingToken(
+    transferableBalancesResponses?.data,
+    stakingTokenMetadata.data,
+    selection,
+  );
```

Comment:
issue: you should convert this to a zustand selector (maybe living in the slice?). That way, you'll get memoization.

Reply:
I specifically wanted to _avoid_ memoization here because **(1)** it wasn’t updating correctly due to edge cases where the `transferableBalancesResponses` and `selection` states remained unchanged across pages, causing the `isStakingToken` boolean to not properly update. Additionally, `hasStakingToken` is a lightweight operation, and I think it’s safer for each page to recalculate this boolean independently.